feature: define Staged Configs (#264)
nginx/documentation: content/includes/nginx-one/staged-config-overview.md

---

> docs:

Reviewer: ```suggestion
```

files:
  - content/nginx-one/how-to/staged-configs/add-staged-config.md
  - content/nginx-one/how-to/staged-configs/edit-staged-config.md

> ---
> It takes time to set up NGINX configuration files can take time. Staged Configurations can help. They work like a draft that uses the features of NGINX One Console. The Staged Configuration does not have to be valid.

Reviewer: ```suggestion
---

It takes time to set up NGINX configuration files can take time. Staged Configurations can help. They work like a draft that uses the features of NGINX One Console. The Staged Configuration does not have to be valid.
```

IIRC Markdownlint adds a blank space between the end of the frontmatter and the start of the content

Reviewer: "It takes time ...can take time" in the same sentence.

Reviewer 2: perhaps something like this?

```
Setting up NGINX configuration files can be time-consuming, but Staged Configurations can help streamline the process. They act as a draft, utilizing the features of the NGINX One Console. The best part is that a Staged Configuration doesn't need to be valid immediately.
```

You can save your work before you push a configuration to an instance of NGINX.

With Staged Configurations, you can use these features of NGINX One Console:

- Automatic syntax checking
- Contextual documentation
- Dynamic analysis of your work in progress
- Suggestions from our AI assistant

> Once you've finished your work and have pushed the changes to an Insance or a Config Sync Group, you are then free to delete that Staged Configuration.

Reviewer: 'once' and 'then' seems redundant

> It takes time to set up NGINX configuration files can take time. Staged Configurations can help. They work like a draft that uses the features of NGINX One Console. The Staged Configuration does not have to be valid.
> You can save your work before you push a configuration to an instance of NGINX.
> 
> With Staged Configurations, you can use these features of NGINX One Console:
> 
> - Automatic syntax checking
> - Contextual documentation
> - Dynamic analysis of your work in progress
> - Suggestions from our AI assistant
> 
> Once you've finished your work and have pushed the changes to an Insance or a Config Sync Group, you are then free to delete that Staged Configuration.

Author: @vrmare I like your approach. I've modified it slightly:

```suggestion

Setting up NGINX configuration files can be time-consuming, but Staged Configurations make the process easier. Think of Staged Configurations as drafts. They give you the flexibility to work without the pressure of needing a valid configuration to save. With Staged Configurations, you can save your progress before pushing changes to an NGINX Instance or Config Sync Group.

This feature leverages the power of NGINX One Console, which allows you to:

- Check syntax automatically
- Access contextual documentation
- Analyze your work in real-time
- Receive suggestions from our AI assistant

Once you're satisfied with your configuration and have pushed it to an Instance or Config Sync Group, you can easily delete the Staged Configuration.
```

Reviewer 2: looks good.

Author: For the record, I'm taking @vrmare 's 👍 as an "approval" from someone who does not (yet) have write access.
